netcdf/nc_double.go

```diff
@@ -92,7 +106,46 @@ func testReadFloat64s(v Var, n uint64) error {
 	}
 	for i := 0; i < int(n); i++ {
 		if val := float64(i + 10); data[i] != val {
-			return fmt.Errorf("data at position %d is %v; expected %v\n", i, data[i], val)
+			return fmt.Errorf("data at position %d is %v; expected %v", i, data[i], val)
+		}
+	}
+	return nil
+}
+
+func testReadFloat64Idx(v Var, n uint64) error {
+	data := make([]float64, n)
+	if err := v.ReadFloat64s(data); err != nil {
+		return err
+	}
+	for i := 0; i < int(n); i++ {
+		shape, _ := v.LenDims()
+		coords, _ := UnravelIndex(uint64(i), shape)
+		expected := float64(i + 10)
+		val, _ := v.ReadIdxFloat64(coords)
+		if val != data[i] {
+			return fmt.Errorf("data at position %v is %v; expected %v", i, val, expected)
+		}
+	}
+	return nil
+}
+
+func testWriteFloat64Idx(v Var, n uint64) error {
+	shape, _ := v.LenDims()
+	ndim := len(shape)
+	coord := make([]uint64, ndim)
+	for i := 0; i < ndim; i++ {
+		for k := 0; k < ndim; k++ {
+			coord[k] = uint64(i)
+		}
+		v.WriteIdxFloat64(coord, float64(i))
+	}
+	for i := 0; i < ndim; i++ {
```

Comment:
This loop should be in `testReadFloat64Idx` but then `testReadFloat64Idx` already does something similar

Reply:
Kind of... I have to test what has been written. So I put the `ReadFloat64Idx` into the `Write` instead of the other way around

Reply:
maybe add a comment explaining why testing Read there